providers/google.go

```diff
@@ -98,17 +101,21 @@ func NewGoogleProvider(p *ProviderData, opts options.GoogleOptions) (*GoogleProv
 		},
 	}
 
+	var file *os.File
 	if opts.ServiceAccountJSON != "" {
-		file, err := os.Open(opts.ServiceAccountJSON)
+		var err error
+		file, err = os.Open(opts.ServiceAccountJSON)
 		if err != nil {
 			return nil, fmt.Errorf("invalid Google credentials file: %s", opts.ServiceAccountJSON)
 		}
+	}
 
```

Comment:
All of this code seems only being used in case the AdminService has to be retrieved via the ServiceAccountJSON and the only place the file variable is actually used is inside the getAdminService function itself. Therefore I would move the logic to the getAdminService function and remove all io.Reader and file reference in this method and the setGroupRestriction.